### PR TITLE
Trip viewer overlay

### PR DESCRIPTION
**Pull request by evansiroky**
This component renders a trip on the base map.

This PR depends on #17.

### CODE REVIEW

File: packages/trip-viewer-overlay/src/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { encodedPolylineType } from "@opentripplanner/core-utils/lib/types";
+import PropTypes from "prop-types";
+import React from "react";
+import { FeatureGroup, MapLayer, Polyline, withLeaflet } from "react-leaflet";
+
+import polyline from "@mapbox/polyline";
+
+class TripViewerOverlay extends MapLayer {
```
Comment (landonreed) [inline packages/trip-viewer-overlay/src/index.js]: Good opportunity to add jsdoc

File: packages/route-viewer-overlay/src/index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  return (
+    routeData &&
+    routeData.patterns &&
+    Object.values(routeData.patterns).every(
+      ptn => typeof ptn.geometry !== "undefined"
+    )
+  );
+};
+
+class RouteViewerOverlay extends MapLayer {
```
Comment (landonreed) [inline packages/route-viewer-overlay/src/index.js]: Good opportunity to add jsdoc

File: packages/trip-viewer-overlay/src/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  render() {
+    const { tripData } = this.props;
+
+    if (!tripData || !tripData.geometry) return <FeatureGroup />;
+
+    const pts = polyline.decode(tripData.geometry.points);
+    return (
+      <FeatureGroup>
+        <Polyline positions={pts} weight={8} color="#00bfff" opacity={0.6} />
```
Comment (landonreed) [inline packages/trip-viewer-overlay/src/index.js]: Should the weight, color, and opacity be props? Maybe we should pass a `polylineProps` that contains the default values shown above that can be overridden?

File: packages/route-viewer-overlay/src/index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    const segments = [];
+    Object.values(routeData.patterns).forEach(pattern => {
+      if (!pattern.geometry) return;
+      const pts = polyline.decode(pattern.geometry.points);
+      segments.push(
+        <Polyline
+          positions={pts}
+          weight={4}
+          color={routeColor}
+          opacity={1}
```
Comment (landonreed) [inline packages/route-viewer-overlay/src/index.js]: [Same comment](https://github.com/opentripplanner/otp-ui/pull/19/files#r363801376) as with trip viewer regarding `polylineProps`

**Comment by evansiroky** [comment]
Just addressed your comments, @landonreed.

File: packages/route-viewer-overlay/src/index.js [hunk 5]
```diff
@@ -16,6 +16,9 @@ const isGeomComplete = routeData => {
   );
 };
 
+/**
+ * An overlay that will display all polylines of the patterns of a route.
```
Comment (landonreed) [inline packages/route-viewer-overlay/src/index.js]: Minor nit pick: isn't this overlay used for patterns from one or many routes?

Comment (evansiroky) [reply packages/route-viewer-overlay/src/index.js]: It's just for one route.

Comment (landonreed) [reply packages/route-viewer-overlay/src/index.js]: Oh, yea, you're totally right. Thanks for confirming.

**Pull request merged**